### PR TITLE
E2E test for support archive log collection.

### PR DESCRIPTION
**Pull request by StefanHauth**
# Description
E2E tests for log collection in support archive (see #1386)

## How can this be tested?
Run E2E tests in total or selectively with
`make manifests/branch test/e2e/supportarchive`

## Checklist
- [x] Unit tests have been updated/added
- [x] PR is labeled accordingly
- [x] I have read and understood the [contribution guidelines](https://github.com/Dynatrace/dynatrace-operator/blob/master/CONTRIBUTING.md)

### CODE REVIEW

File: test/support_archive/support_archive_test.go [hunk 1]
```diff
@@ -58,22 +60,32 @@ func SupportArchiveExecution(t *testing.T) features.Feature {
...
+		requiredFiles = append(requiredFiles, "operator-version.txt")
+		requiredFiles = append(requiredFiles, getRequiredLogFiles(t, ctx, environmentConfig)...)
+
 		tarReader := tar.NewReader(zipReader)
+
 		hdr, err := tarReader.Next()
-		require.NoError(t, err)
-		assert.Equal(t, "operator-version.txt", hdr.Name)
+		for err == nil {
+			idx := slices.IndexFunc(requiredFiles, func(file string) bool { return file == hdr.Name })
```
Comment (luhi-DT) [inline test/support_archive/support_archive_test.go:77]:
```suggestion
			index := slices.IndexFunc(requiredFiles, func(file string) bool { return file == header.Name })
```

File: test/support_archive/support_archive_test.go [hunk 2]
```diff
@@ -58,22 +60,32 @@ func SupportArchiveExecution(t *testing.T) features.Feature {
...
+			assert.NotEqualf(t, -1, idx, "Found unexpected file %s.", hdr.Name)
 
-		resultString := make([]byte, hdr.Size)
-		resultLen, err := tarReader.Read(resultString)
-		require.Equal(t, io.EOF, err)
-		assert.Equal(t, hdr.Size, int64(resultLen))
+			if idx != -1 {
```
Comment (luhi-DT) [inline test/support_archive/support_archive_test.go:80]:
```suggestion
			if index != -1 {
```

File: test/support_archive/support_archive_test.go [hunk 3]
```diff
@@ -101,6 +113,20 @@ func executeSupportArchive(ctx context.Context, t *testing.T, environmentConfig
...
 }
 
+func getRequiredLogFiles(t *testing.T, ctx context.Context, envConfig *envconf.Config) []string {
+	resources := envConfig.Client().Resources()
+	opPods := operator.Get(t, ctx, resources)
+
+	requiredLogs := make([]string, 0)
+
+	for _, pod := range opPods.Items {
+		for _, cont := range pod.Spec.Containers {
```
Comment (luhi-DT) [inline test/support_archive/support_archive_test.go:123]:
```suggestion
		for _, container := range pod.Spec.Containers {
```

File: test/support_archive/support_archive_test.go [hunk 4]
```diff
@@ -58,22 +60,32 @@ func SupportArchiveExecution(t *testing.T) features.Feature {
...
 	return func(ctx context.Context, t *testing.T, environmentConfig *envconf.Config) context.Context {
-		result := executeSupportArchive(context.Background(), t, environmentConfig, "--stdout")
+		result := executeSupportArchive(ctx, t, environmentConfig, "--stdout")
 		require.NotNil(t, result)
 
 		zipReader, err := gzip.NewReader(result.StdOut)
 		require.NoError(t, err)
 
+		requiredFiles := make([]string, 0)
+		requiredFiles = append(requiredFiles, "operator-version.txt")
```
Comment (luhi-DT) [inline test/support_archive/support_archive_test.go:70]: Can this be a const?

**Review by luhi-DT — CHANGES_REQUESTED** [review]
Comments added

File: test/support_archive/support_archive_test.go (hunk 1 above)
Comment (mjgrzybek) [reply test/support_archive/support_archive_test.go:77]: `idx` is fine IMHO.

**Review by luhi-DT — APPROVED** [review]

**Pull request merged**